packages/headless/src/controllers/commerce/context/cart/headless-cart.ts

```diff
@@ -84,20 +85,22 @@ export interface Cart extends Controller {
    * - Setting `quantity` to a positive number creates the item in the cart with the specified `productName`,
    * `pricePerUnit`, and `quantity`.
    *
-   * If the specified `quantity` is equivalent to the current quantity of the item in the cart, no analytics event is
-   * emitted. Otherwise, the method emits an `ec.cartAction` event with the appropriate action (`add` or `remove`).
-   *
    * @param item - The cart item to create, update, or delete.
    */
   updateItem(item: CartItem): void;
```

Comment:
Another approach could be to add a second optional `track` parameter, defaulting to `false`.
The key theme to solve the different issues in the PR description is allowing implementers to opt-in to the behavior, whereas today they do not have such control,

Reply:
I think it would feel weird to have a boolean flag to optionally trigger relay events. Other events triggered through headless don't have a similar mechanism. I think I prefer splitting the logic in two methods 🤔

Reply:
Is it ok to have two different api styles?

If not, we could use two methods, but to name them explicitly. e.g. `updateItem` and `updateItemAndTrackCartAction`
If one needs to sync state, they use the first one, if they want to update state and track, they use the second.

We'd also do the same for `empty`? `emptyAndTrackCartAction` ...

If the naming is too verbose, another idea could be to create property `withAnalytics` that exposes these methods. e.g.

```
cart.withAnalytics.updateItem()
cart.withAnalytics.empty()
``` 
With such a pattern, would `purchase` need to move under the `withAnalytics` object?

Reply:
> With such a pattern, would purchase need to move under the withAnalytics object?

It might be relevant to mention here, but I don't think `purchase` is fully decoupled. If I remember correctly, when you call `purchase`, you empty the cart as well. In that sense, they're would be two patterns the implementer might have to choose when implementing their solution; one with analytics and one without.

Reply:
Great point, confirming that `purchase` [empties the cart](https://github.com/coveo/ui-kit/blob/master/packages/headless/src/features/commerce/context/cart/cart-slice.ts#L41)

This suggests that the current PR is not right. The supported modes on the cart controller might be:
- state-only
- state + analytics
but not analytics-only